src/astro/files/operators/files.py

```diff
@@ -0,0 +1,36 @@
+from typing import Any
+
+from airflow.decorators.base import get_unique_task_id
+from airflow.models import BaseOperator
+from airflow.utils.context import Context
+
+from astro.files.base import File
+from astro.files.locations import create_file_location
+
+
+class ListFileOperator(BaseOperator):
+    """List the file available at path and storage
+
+    :param task_id: The task id for uniquely identify a task in a DAG
+    :param path: A path pattern for which you want to get a list of file
+    :param conn_id: connection id for the services
+    """
+
+    template_fields = ("path", "conn_id")
+
+    def __init__(self, path: str, conn_id: str, task_id: str = "", **kwargs):
+        task_id = task_id or get_unique_task_id(
+            "get_file_list", dag=kwargs.get("dag"), task_group=kwargs.get("task_group")
+        )
```

Comment:
This should be set on `ListFileOperator` creation from outside i.e. inside the `get_file_list` function. With that being said, you can then also remove `task_id: str = ""` from the `__init__`.

Reply:
it was earlier in `get_file_list` but we decided to move it to the operator.

Reply:
Okay. Could you explain why you decided to move it? Because how should `ListFileOperator` know that it gets called within `get_file_list` function? Only then, it makes sense to name the task `get_file_list`. In all other cases `get_file_list` here as task id does not make sense.

Reply:
just curious, why do ``ListFileOperator`` should know who calls it?
If task_id logic would be in operator then ``get_file_list`` would be a little bit lite and also below code would work I.e you can directly call ListFileOperator without passing `task_id`

```
op1 = ListFileOperator(conn, path)
op2 = ListFileOperator(conn, path)
```

but if you move it in `get_file_list` then you will have pass task_id if you directly use ``ListFileOperator``

regarding name  ``get_file_list`` is just a thin wrapper to return response of ListFileOperator task.

Reply:
> just curious, why do ListFileOperator should know who calls it?

That's what I mean. It should not, but to me it seems that it is aware of who is intended to be the caller as `get_file_list` matches the function name of the function that calls it i.e. `get_file_list`. If it is not clear, we can also quickly jump on a call :)

> but if you move it in get_file_list then you will have pass task_id if you directly use ListFileOperator

Isn't `task_id` a requirement for every operator inheriting from `BaseOperator`, if so you don't have to explicitly add it as an argument.